src/agent/src/rpc.rs

```diff
@@ -1737,6 +1774,137 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
     }
 }
 
+fn pull_image_from_registry(
+    image: &str,
+    cid: &str,
+    source_creds: &Option<&str>,
+    use_aa: bool,
+    agent_started: bool,
+) -> Result<()> {
+    let source_image = format!("{}{}", "docker://", image);
+
+    let manifest_path = format!("/tmp/{}/image_manifest", cid);
+    let target_path_manifest = format!("dir://{}", manifest_path);
+
+    // Define the target transport and path for the OCI image, without signature
+    let oci_path = format!("/tmp/{}/image_oci:latest", cid);
+    let target_path_oci = format!("oci://{}", oci_path);
+
+    fs::create_dir_all(&manifest_path)?;
+    fs::create_dir_all(&oci_path)?;
+
+    info!(sl!(), "Attempting to pull image {}...", &source_image);
+
+    let mut pull_command = Command::new(SKOPEO_PATH);
+    pull_command
+        .arg("--insecure-policy")
```

Comment:
Please add a comment explaining why this option is needed.

Reply:
Again, not part of my patch.